bolt/lib/Passes/ReorderFunctions.cpp

```diff
@@ -269,9 +266,11 @@ Error ReorderFunctions::readFunctionOrderFile(
 }
 
 Error ReorderFunctions::runOnFunctions(BinaryContext &BC) {
+  if (opts::ReorderFunctions == RT_NONE) {
```

Comment:
I tried your change and it looks like there's quite a few tests failing because they depend on the behavior that is currently encoded (to run this logic even if we have RT_NONE). Perhaps this change is not as trivial and it seems we are overlooking some necessary steps here when we have RT_NONE.

If you want to check, you can run ninja check-bolt to see the tests that fail. I suggest we leave this as is and classify reorder functions as mandatory, even when no order is specified.

Reply:
Interesting. Thanks for all the pointers. For the sake of simplicity I'll make this PR just add the option to skip instruction shortening.